apps/onnx/onnx_converter_generator.cc

```diff
@@ -0,0 +1,95 @@
+#include "onnx_converter.h"
+#include <Halide.h>
+#include <map>
+#include <fstream>
+
+namespace {
+
+// TODO: This should be an onnx_converter API.
+Halide::Type getHalideType(const Tensor &tensor) {
+    switch (tensor.type) {
+    case onnx::TensorProto_DataType_FLOAT:
+        return Halide::Float(32);
+    case onnx::TensorProto_DataType_DOUBLE:
+        return Halide::Float(64);
+    case onnx::TensorProto_DataType_INT8:
+        return Halide::Int(8);
+    case onnx::TensorProto_DataType_INT16:
+        return Halide::Int(16);
+    case onnx::TensorProto_DataType_INT32:
+        return Halide::Int(32);
+    case onnx::TensorProto_DataType_UINT8:
+        return Halide::UInt(8);
+    case onnx::TensorProto_DataType_UINT16:
+        return Halide::UInt(16);
+    case onnx::TensorProto_DataType_UINT32:
+        return Halide::UInt(32);
+    case onnx::TensorProto_DataType_INT64:
+        return Halide::Int(64);
+    case onnx::TensorProto_DataType_BOOL:
+        return Halide::Bool();
+    default:
+        throw std::domain_error("Unsupported or unknown target type");
```

Comment:
Halide doesn't use exceptions for error handling

Reply:
onnx_converter library use exception. should we keep exceptions here as well for consistency ?

Reply:
I didn't review that code, or I would have made the same comment there.

I'll defer to @abadams here, but as of now, the only place we use exceptions in Halide are in the Python wrapper code (where it's essentially required for PyBind11 usage); AFAIK there is no other place in any of our codebase that uses C++ exceptions, and while I'm not intrinsically opposed to them, consistency in the codebase is a good thing. If we want to allow C++ exception usage in Halide code, we should do so deliberately and by consensus.

Reply:
onnx_converter also going to have a Pybind11 python extension in a follow up PR. so I thought keeping exceptions here as well.